### PR TITLE
[FIX] web: use optionalActiveFields for direct export

### PR DESCRIPTION
**Pull request by hubvd**
Steps to reproduce
==================

- Go to project
- Switch to the list view
- Remove some optional fields
- Click on the export all button next to the "New Button"

The optional fields configuration are ignored and all fields are exported.

Cause of the issue
==================

The `optionalActiveFields` are not taken into account when computing the `defaultExportList`.

This was the case in 15.0

opw-3452459

### CODE REVIEW

File: addons/web/static/src/views/list/list_renderer.js [hunk 1]
```diff
@@ -1595,6 +1598,9 @@ export class ListRenderer extends Component {
 
     async toggleOptionalField(fieldName) {
         this.optionalActiveFields[fieldName] = !this.optionalActiveFields[fieldName];
+        if (this.props.optionalFieldsChanged) {
+            this.props.optionalFieldsChanged(this.optionalActiveFields);
```
Comment (Iucapad) [inline addons/web/static/src/views/list/list_renderer.js:1602]: Can you please rename to `onOptionalFieldsChanged`?

**Review by Iucapad — APPROVED** [review]
Looks good to me.

Just one small comment regarding the name of the props

File: addons/web/static/src/views/list/list_renderer.js (hunk 1 above)
Comment (hubvd) [reply addons/web/static/src/views/list/list_renderer.js]: Done :+1:

**Comment by Iucapad** [comment]
@robodoo r+

**Pull request closed without merging**